source/d3d8to9_base.cpp

```diff
@@ -202,13 +204,21 @@ HRESULT STDMETHODCALLTYPE Direct3D8::CreateDevice(UINT Adapter, D3DDEVTYPE Devic
 
 	IDirect3DDevice9 *DeviceInterface = nullptr;
 
-	const HRESULT hr = ProxyInterface->CreateDevice(Adapter, DeviceType, hFocusWindow, BehaviorFlags, &PresentParams, &DeviceInterface);
+	HRESULT hr = ProxyInterface->CreateDevice(Adapter, DeviceType, hFocusWindow, BehaviorFlags, &PresentParams, &DeviceInterface);
+
+	if (hr == D3DERR_DEVICELOST && pCurrentDeviceInterface)
+	{
+		pCurrentDeviceInterface->Reset(&CurrentPresentParams);
```

Comment:
Great idea!  But I just tried it and I see a crash sometimes when changing resolution if I do that.  Not sure why but it could be related to either the `hDeviceWindow` or maybe even one of the `D3DFORMAT` parameters (such as `BackBufferFormat` or `AutoDepthStencilFormat`).  Too bad.